scripts/build-toolchains.sh

```diff
@@ -147,5 +147,5 @@ cd "$RDIR"
 
 # create general env.sh
 echo "# line auto-generated by build-toolchains.sh" >> env.sh
-echo "source \$( realpath \$(dirname "\${BASH_SOURCE[0]:-\${\(%\):-%x}}") )/env-$TOOLCHAIN.sh" >> env.sh
+echo "source $PWD/env-$TOOLCHAIN.sh" >> env.sh
```

Comment:
I feel like if it were this simple it would have been done before.

Reply:
In general, you can't always expect users to `cd` into the chipyard top directory before sourcing `env.sh`.

Reply:
this will not result in a bug. $PWD resolved in the current shell after i `cd $CHIPYARD_DIR`

Reply:
i'm surprised it was not done this way before

Reply:
If the path contains any whitespace, naively expanding `$PWD` at generation time breaks the `source` command due to word splitting.

Reply:
that bug existed in the original code too

Reply:
Can't you just put some escaped `\"` quotes in to fix that?

Reply:
yes

Reply:
i used the fix proposed by @a0u